pkg/codegen/pcl/binder_test.go

```diff
@@ -448,14 +442,35 @@ resource randomPet "random:index/randomPet:RandomPet" {
   prefix = data
 }`
 
-	lenientProgram, lenientDiags, lenientError :=
-		ParseAndBindProgram(t, source, "program.pp", pcl.SkipResourceTypechecking)
+	lenientProgram, lenientDiags, lenientError := ParseAndBindProgram(t, source, "program.pp", pcl.SkipResourceTypechecking)
 	require.NoError(t, lenientError)
 	assert.False(t, lenientDiags.HasErrors(), "There are no errors")
 	assert.NotNil(t, lenientProgram)
 
-	strictProgram, _, strictError :=
-		ParseAndBindProgram(t, source, "program.pp")
+	strictProgram, _, strictError := ParseAndBindProgram(t, source, "program.pp")
+	assert.NotNil(t, strictError, "Binding fails in strict mode")
+	assert.Nil(t, strictProgram)
+}
+
+func TestAssigningUnknownPropertyFromKnownResourceWhenSkippingResourceTypeChecking(t *testing.T) {
+	t.Parallel()
+	// here the resource declaration is correctly typed but the output mainId references an unknown property
```

Comment:
comment needs updating, also we don't need the "mainId" output for this test.

Reply:
Yeah my bad! Updated the comment